backend/src/main/java/capstone/backend/services/ProductService.java

```diff
@@ -36,26 +40,32 @@ public ProductDTO getProductDetails(Long id) {
     }
 
     public ProductDTO createProduct(ProductDTO product) throws EntityWithThisIdAlreadyExistException {
-        if (product.getId() != null && repo
-                .findById(product.getId())
-                .isPresent()) {
+        if (productExists(product)) {
             throw new EntityWithThisIdAlreadyExistException(String.format("Product %s already has the id %d", product.getName(), product.getId()));
         }
         return ProductMapper.mapProductWithDetails(repo.
                 save(ProductMapper.mapProduct(product)));
     }
 
     public ProductDTO editProduct(ProductDTO product) {
-        if (repo
-                .findById(product.getId())
-                .isEmpty()) {
+        if (!productExists(product)) {
             throw new EntityNotFoundException(String.format("Couldn't find a product with the id %d", product.getId()));
```

Comment:
"Couldn't find any product with id %d"

Reply:
%d is the place holder for any number formats in a string format

Reply:
I was just suggesting a rephrase

Reply:
ah, alrigth. focus on any. didn't notice earlier